datafusion/core/tests/dataframe/mod.rs

```diff
@@ -1298,6 +1298,96 @@ async fn unnest_aggregate_columns() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn unnest_array_agg() -> Result<()> {
```

Comment:
I verified that this test covers the code in this PR by running it without the code changes locally. It fails as expected 💯 

```

---- dataframe::unnest_array_agg stdout ----
Error: ArrowError(ComputeError("concat requires input of at least one array"))

```